### PR TITLE
Add inline example for blockmean

### PR DESCRIPTION
**Pull request by michaelgrund**
**Description of proposed changes**

Addresses https://github.com/GenericMappingTools/pygmt/issues/1686

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If wrapping a new module, open a 'Wrap new GMT module' issue and submit reasonably-sized PRs.
- [ ] If adding new functionality, add an example to docstrings or tutorials.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

File: pygmt/src/blockm.py [hunk 1]
```diff
@@ -147,6 +147,16 @@ def blockmean(data=None, x=None, y=None, z=None, outfile=None, **kwargs):
           is not set.
         - None if ``outfile`` is set (filtered output will be stored in file
           set by ``outfile``).
+
+    Examples
+    --------
```
Comment (willschlitzer) [inline pygmt/src/blockm.py:152]:
```suggestion
    Example
    -------
```

I think it should just be "Example" as there is only one block of code

**Review by seisman — APPROVED** [review]

**Review by willschlitzer — APPROVED** [review]

File: pygmt/src/blockm.py [hunk 2]
```diff
@@ -147,6 +147,16 @@ def blockmean(data=None, x=None, y=None, z=None, outfile=None, **kwargs):
           is not set.
         - None if ``outfile`` is set (filtered output will be stored in file
           set by ``outfile``).
+
+    Example
+    -------
+    >>> import pygmt  # doctest: +SKIP
+    >>> # Load a table of ship observations of bathymetry off Baja California
+    >>> data = pygmt.datasets.load_sample_bathymetry()
```
Comment (meghanrjones) [inline pygmt/src/blockm.py:155]:
```suggestion
    >>> data = pygmt.datasets.load_sample_data(name='bathymetry')
```
To avoid the deprecation warning.

**Comment by crystalratna** [comment]
I was searching how to add -C (Use the center of the block as the output location) in pygmt.blockmean command line.
Thank you.

**Comment by yvonnefroehlich** [comment]
> I was searching how to add -C (Use the center of the block as the output location) in pygmt.blockmean command line. Thank you.

Thanks @crystalratna for trying out PyGMT and apologizes for the late response!

Currently, there is no alias available for `-C` for the `pygmt.blockmean` function in PyGMT. Maybe you can try `C=True` (related to https://docs.generic-mapping-tools.org/dev/blockmean.html#c), but I am not sure about this.
Ping @GenericMappingTools/pygmt-maintainers for help on this, please.

If this comment is meant as a request for adding an alias for `-C` for the `pygmt.blockmean` function, please open a separate issue.  Probably this comment was overlooked as it is written in a merged PR.

Please see the extended line example below:
```python
# sourec: https://www.pygmt.org/dev/api/generated/pygmt.blockmean.html#pygmt.blockmean
# last acces: 2022/11/06
import pygmt
# Load a table of ship observations of bathymetry off Baja California
data = pygmt.datasets.load_sample_data(name="bathymetry")
# Calculate block mean values within 5 by 5 minute bins
data_bmean = pygmt.blockmean(
    data=data,
	region=[245, 255, 20, 30],
	spacing="5m",
	C=True,  # Use the center of the block as the output location
)
```

**Comment by crystalratna** [comment]
Thank you. I tried and it seems to work.

**Pull request merged**